src/lib.rs

```diff
@@ -311,9 +311,13 @@ pub unsafe extern "C" fn wgpuSurfaceGetPreferredFormat(
     surface: id::SurfaceId,
     adapter: id::AdapterId,
 ) -> native::WGPUTextureFormat {
-    let preferred_format = match wgc::gfx_select!(adapter => GLOBAL.surface_get_preferred_format(surface, adapter))
+    let preferred_format = match wgc::gfx_select!(adapter => GLOBAL.surface_get_supported_formats(surface, adapter))
     {
-        Ok(format) => conv::to_native_texture_format(format),
+        Ok(formats) => conv::to_native_texture_format(
+            *formats
+                .first()
+                .expect("Could not get preferred swap chain format"),
+        ),
```

Comment:
For reference: https://github.com/gfx-rs/wgpu/blob/0d4d3f4ceb0c07de38064533c3409a2f8c1d8d95/wgpu/src/lib.rs#L3448-L3454  the first format is the preferred one.